subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,609 @@
+                    "node1", ":producer:producer",
+                    []
+                ),
+                transformMatcher(
+                    "node2", ":consumer", [buildPath: ":", projectPath: ":producer"], [artifactType: "jar", color: "red"],
+                    ["node1"]
+                ),
+                transformMatcher(
+                    "node3", ":consumer", [buildPath: ":", projectPath: ":producer"], [artifactType: "jar", color: "green"],
+                    ["node2"]
+                ),
+                taskMatcher(
+                    "node4", ":consumer:resolve",
+                    ["node3"]
+                ),
+            ]
+        )
+
+        def executeTransformationOps = buildOperations.all(ExecuteScheduledTransformationStepBuildOperationType)
+        def executeTransformationOp1 = executeTransformationOps[0]
+        with(executeTransformationOp1.details) {
+            transformerName == "MakeRed"
+            subjectName == "producer.jar (project :producer)"
+            with(transformationIdentity) {
+                nodeType == "ARTIFACT_TRANSFORM"
+                buildPath == ":"
+                projectPath == ":consumer"
+                targetVariant.componentId == [buildPath: ":", projectPath: ":producer"]
+                targetVariant.attributes == [color: "red", artifactType: "jar"]
+                targetVariant.capabilities == [[group: "colored", name: "producer", version: "unspecified"]]
+                artifactName == "producer.jar"
+                dependenciesConfigurationIdentity == null
+            }
+            transformType == "MakeRed"
+            sourceAttributes == [color: "blue", artifactType: "jar"]
+            fromAttributes == [color: "blue"]
+            toAttributes == [color: "red"]
+        }
+
+        def executeTransformationOp2 = executeTransformationOps[1]
+        with(executeTransformationOp2.details) {
+            transformerName == "MakeGreen"
+            subjectName == "producer.jar (project :producer)"
+            with(transformationIdentity) {
+                nodeType == "ARTIFACT_TRANSFORM"
+                buildPath == ":"
+                projectPath == ":consumer"
+                targetVariant.componentId == [buildPath: ":", projectPath: ":producer"]
+                targetVariant.attributes == [color: "green", artifactType: "jar"]
+                targetVariant.capabilities == [[group: "colored", name: "producer", version: "unspecified"]]
+                artifactName == "producer.jar"
+                dependenciesConfigurationIdentity == [buildPath: ":", projectPath: ":consumer", name: "resolver"]
+            }
+            transformType == "MakeGreen"
+            sourceAttributes == [color: "red", artifactType: "jar"]
+            fromAttributes == [color: "red"]
+            toAttributes == [color: "green"]
+        }
+    }
+
+    def checkExecutionPlanMatchingDependencies(List<PlannedNode> plannedNodes, List<NodeMatcher> nodeMatchers) {
```

Comment:
```suggestion
    void checkExecutionPlanMatchingDependencies(List<PlannedNode> plannedNodes, List<NodeMatcher> nodeMatchers) {
```

❌  So you don't need to return true at the end.

Reply:
Thanks, fixed!